### PR TITLE
tune interface to support predicting loaded built-in algorithms; remove stale task in cache

### PR DESCRIPTION
**Pull request by ylwu-amzn**
Signed-off-by: Yaliang Wu <ylwu@amazon.com>

### Description
tune interface to support predicting loaded built-in algorithms;
remove stale task in cache
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: ml-algorithms/src/main/java/org/opensearch/ml/engine/ModelHelper.java [hunk 1]
```diff
@@ -86,13 +86,13 @@ private void verifyModelZipFile(String modelZipFilePath) throws IOException {
                 String fileName = ((ZipEntry) zipEntries.nextElement()).getName();
                 if (fileName.endsWith(PYTORCH_FILE_EXTENSION) || fileName.endsWith(ONNX_FILE_EXTENSION)) {
                     if (hasModelFile) {
-                        throw new IllegalArgumentException("Find multiple model files");
+                        throw new IllegalArgumentException("Find multiple model files, but expected only one");
                     }
                     hasModelFile = true;
                 }
                 if (fileName.equals(TOKENIZER_FILE_NAME)) {
```
Comment (dhrubo-os) [inline ml-algorithms/src/main/java/org/opensearch/ml/engine/ModelHelper.java:93]: how can we have two files with the same name `tokenizer.json`?

File: plugin/src/main/java/org/opensearch/ml/task/MLTaskManager.java [hunk 2]
```diff
@@ -384,4 +389,39 @@ public boolean containsModel(String modelId) {
...
+            MLTask mlTask = entry.getValue().getMlTask();
+            if (mlTask.getTaskType() == MLTaskType.LOAD_MODEL && mlTask.getState() != MLTaskState.CREATED) {
+                runningLoadModelTaskIds.add(entry.getKey());
+            }
+        }
+        return runningLoadModelTaskIds.toArray(new String[0]);
+    }
+
+    public void syncRunningLoadModelTasks(Map<String, Set<String>> runningLoadModelTasks) {
+        Instant ttlEndTime = Instant.now().minus(120, ChronoUnit.SECONDS);
```
Comment (Zhangxunmt) [inline plugin/src/main/java/org/opensearch/ml/task/MLTaskManager.java:405]: Is this 2 mins too short? If possible, why not use a longer time? It does no harm to keep a task for 10 mins?

File: plugin/src/main/java/org/opensearch/ml/action/syncup/TransportSyncUpOnNodeAction.java [hunk 3]
```diff
@@ -141,9 +144,13 @@ private MLSyncUpNodeResponse createSyncUpNodeResponse(MLSyncUpNodesRequest loadM
             mlModelManager.syncModelRouting(modelRoutingTable);
```
Comment (Zhangxunmt) [inline plugin/src/main/java/org/opensearch/ml/action/syncup/TransportSyncUpOnNodeAction.java:144]: Is it useful to add a log info here showing an import action is being taken like this "clear Routing Table"? Same for remove worker nodes, add work nodes, etc?

**Review by Zhangxunmt — APPROVED** [review]

**Review by dhrubo-os — APPROVED** [review]

File: plugin/src/main/java/org/opensearch/ml/task/MLTaskManager.java (hunk 2 above)
Comment (ylwu-amzn) [reply plugin/src/main/java/org/opensearch/ml/task/MLTaskManager.java:405]: Yes, good point. I used 2 minutes for quick testing. I think we do need to increase to a longer TTL. 10 mins looks good. Will change this part in next PR.

File: ml-algorithms/src/main/java/org/opensearch/ml/engine/ModelHelper.java (hunk 1 above)
Comment (ylwu-amzn) [reply ml-algorithms/src/main/java/org/opensearch/ml/engine/ModelHelper.java:93]: Good point, planned to check nested subfolder, should be `endsWith`. Now the code doesn't support nested subfolders in model zip. So we can remove this check now. Will fix this in next PR.

**Pull request merged**